### PR TITLE
Update dev dependencies

### PR DESCRIPTION
**Pull request by jameshilliard**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -98,7 +98,7 @@ jobs:
       - name: ruff
         if: matrix.run_lint == true
         run: |
-          ruff .
+          ruff check .
```
Comment (janiversen) [inline .github/workflows/ci.yml:101]: This a mixture of CI and version bumps, as written earlier:
- CI changes are in seperate PR
- version bumps are in a seperate PR.

This is done for several reasons:
- mixed changes are harder to bisect
- version bumps need to be clearly identifiable in the changelog for users
- CI changes and version bumps needs extra handling in github actions

**Review by janiversen — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jameshilliard) [reply .github/workflows/ci.yml:101]: Split the CI change out to #2242.

Comment (janiversen) [reply .github/workflows/ci.yml:101]: This one still contains the CI part.

Comment (jameshilliard) [reply .github/workflows/ci.yml]: Removed it from this one(although tests fail again without that).

Comment (janiversen) [reply .github/workflows/ci.yml]: The test should not be affected by the 3rd party version numbers, but I will ask CI to run on this PR (due to CI setup it is more than just starting CI).

**Review by janiversen — APPROVED** [review]
Lgtm

**Comment by alexrudd2** [comment]
The change from `ruff .` to `ruff check .` is not a change in CI functionality, it's a required migration for Ruff `0.5.0`.

See documentation of the change here: https://astral.sh/blog/ruff-v0.5.0#removed-deprecated-features

**Comment by janiversen** [comment]
@alexrudd2 I agree, I read the same, but nonetheless CI changes are in a separate CI, which are merged.

**Pull request merged**